plugin/Field/Collection/FieldLoaderEstateRegionValues.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+/**
+ *
+ *    Copyright (C) 2019 onOffice GmbH
+ *
+ *    This program is free software: you can redistribute it and/or modify
+ *    it under the terms of the GNU Affero General Public License as published by
+ *    the Free Software Foundation, either version 3 of the License, or
+ *    (at your option) any later version.
+ *
+ *    This program is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *    GNU Affero General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Affero General Public License
+ *    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+declare (strict_types=1);
+
+namespace onOffice\WPlugin\Field\Collection;
+
+use Generator;
+use onOffice\SDK\onOfficeSDK;
+use onOffice\WPlugin\API\APIClientActionGeneric;
+use onOffice\WPlugin\API\APIEmptyResultException;
+use onOffice\WPlugin\Language;
+use onOffice\WPlugin\Region\Region;
+use onOffice\WPlugin\Region\RegionController;
+use onOffice\WPlugin\Region\RegionFilter;
+use onOffice\WPlugin\SDKWrapper;
+use onOffice\WPlugin\Types\FieldTypes;
+
+/**
+ *
+ */
+
+class FieldLoaderEstateRegionValues
+	implements FieldLoader
+{
+	/** @var SDKWrapper */
+	private $_pSDKWrapper;
+
+	/** @var RegionController */
+	private $_pRegionController;
+
+	/** @var RegionFilter */
+	private $_pRegionFilter;
+
+
+	/**
+	 * @param SDKWrapper $pSDKWrapper
+	 * @param RegionController $pRegionController
+	 * @param RegionFilter $pRegionFilter
+	 */
+	public function __construct(
+		SDKWrapper $pSDKWrapper,
+		RegionController $pRegionController,
+		RegionFilter $pRegionFilter)
+	{
+		$this->_pSDKWrapper = $pSDKWrapper;
+		$this->_pRegionController = $pRegionController;
+		$this->_pRegionFilter = $pRegionFilter;
+	}
+
+	/**
+	 * @return Generator
+	 * @throws APIEmptyResultException
+	 */
+	public function load(): Generator
```

Comment:
This does the same as `FieldLoaderGeneric` and adds the region loading on top. I think, this should either _only load the regions_ or _use `FieldLoaderGeneric`_ so that the code isn't duplicated

Reply:
It'd be nice if there was a test for the new class as well.